### DISCUSSION TITLE
absolute positioning of images on a slide

### DISCUSSION THREAD

**Discussion by abotzki** [LiaScript/LiaScript#151] [Q&A] [answered]
Hi, I was wondering whether someone found a solution to a slide layout where one would apply absolute positions.

As far as I see you could apply style elements like shown [here](https://liascript.github.io/course/?data:text/plain;base64,IyBFeGFtcGxlIGFic29sdXRlIHBvc2l0aW9ucwoKIVtdKGh0dHBzOi8vcXVhcnRvLm9yZy9kb2NzL3ByZXNlbnRhdGlvbnMvcmV2ZWFsanMvZGVtby9taW5pL2ltYWdlcy9raXR0ZW4tNDAwLTM1MC5qcGVnKTwhLS0gc3R5bGU9InRvcDogMTcwcHg7IGxlZnQ6IDMwcHg7IHdpZHRoOiA0MDBweDsgaGVpZ2h0OiA0MDBweDsiLS0+ ). Unfortunately, the top and left style element is ignored.

But I think that we'd need a class like this one for this image to make it work. 
```
.absolute {
  position: absolute;
  display: block;
}
```

Any suggestion how to get this behaviour? Thanks a lot.

**Comment by andre-dietrich** [comment, marked as answer]
You can also directly set these attributes within the inline-style attribute, but of course, you can also define classes. I think simply adding `position: absolute` to your HTML-comment will do the trick. However, I add some examples how to change image settings, that you can [Try out here - LiveEditor](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA71V32/bNhB+519xVRCgHSLLaBYU8JIgQfewYBlQLNlTUcC0eLZYU6RGnpoZw/733ZGyk2bFBmxY/WBLd8eP9+P7zucv6lpdJdo5VLP8Y2rb6w3C7wr482ANdQs4nc+H377LliEkSzb4BehVCm4kLPZVIAr9Al7Pj4vB4Zr45Bm//qGu0BtV15dKHcFNxn834Sh1a/VdG+1AQNFiAgpgDXqy6x1QhxAG4pQcpFY76zewDhG0h5zmCTx0tu2gR+3TQqlvwK7BEliBCZI+5njBSW1E9JNXwNCACQ9+fyhBh3bTTYdhZTf/eFTi3pwdQ1g/DSswz3NJXIOb3n2gCUaiGKZjY6sTwi6M/OBBuxSgdbbdQvB8C37CuCs1y6XonY78yHCz2QyUevH+R0uE/sPLjmhIi6b5ddSRwizETWNCm5ohYuKuaml6aiIDavcxNQb70PTW2yaDp2abcepv5/P69Gw++zjg5hWP7QiujZH26wMFlMqjTKAjwkonyxU5TtLzmBDQYc/3pROubuTvQ2XDwGSQCf5w/9Nt3YZewnhU1LFxB1vLXm5o9g466h4JYx5Jy62IweVZCQw7fYszuFln9AfNOBLWaZ97kw6pFr/lAXT6U+5gQsozW/6V0Mv/u53nrDrIaruoPlPYl9R1UNZpjiiyes3PFRRBHcEvSQbTOp0SJqWuuUxBz8Y9l1YI2hjmrRU6vb27q6f4xVepNl92UT1dMY/5f7/zurftoXpOnLmVlZJzZ3uyK5fnNrJItCiNF0ZNeiM2U85n9vF3OzpNWUrxcXkgEcMyILPryT0zuGcScHWjI2HdHjoLXXi60ixBvuNxS+kkUbyr/OYkLytv+G5CIbqmouXihtSF0RnpPUXkEFMOP2Ilb5nHVNSxp2av41aUXSTBWDwLQVve3lzfvf355t39Yvn5opCmLEUw7JB+2IgtcTMGzqLQPKvJB4OJ94XsyYl351O5YaRhpImOFRhca+7IRXU6r7jA7Iqiqgri6OvAsru8yvbzpiBcwrHA7sn6HLfY/wNw4f1zWLH+a9A92KREY9Pg9G4BKxfabQV9MPwXdFGtucVYXaon7YevvyBK/i+X+XX56vjvd8U+urzn8NLAvUPe2Jw1uFSHtvwJGyu8VRAIAAA=)

``` markdown
<!--
@style
.styled-image {
    width: 300px;
    position: absolute;
    bottom: 20%;
    left: 35%;
}
@end
-->

# Image Position

LiaScript tries to identify the optimal scaling for an image, which means:

* if it is too wide for the screen it is scaled down
* if its height is to big for the screen it is scaled down to 75% of the screen height
* if it is too small it is not scaled
* in this case you can also click onto every image to enlarge it ... 

![Kitten](https://quarto.org/docs/presentations/revealjs/demo/mini/images/kitten-400-350.jpeg)

## Adding a position

Images are basically inline elements, thus, you can append an HTML-comment with any kind of
HTML-parameters to control its appearance. If you want to change its position you will have to
set the `position: absolute`

![Kitten](https://quarto.org/docs/presentations/revealjs/demo/mini/images/kitten-400-350.jpeg)<!--
style="width: 300px; position: absolute; bottom: 30px; left: 20px"
-->

## Using classes

All styling can also be added into CSS-classes:

![Kitten](https://quarto.org/docs/presentations/revealjs/demo/mini/images/kitten-400-350.jpeg)<!-- 
class="styled-image"
-->

## Dynamic positioning

It is also possible to use a script-tag to dynamically calculate your optimal settings and positioning.
The result of a script is send back to LiaScript as a string, to indicate, that this string should be
treated as a LiaScript snippet, you have to mark it with a starting `LIASCRIPT:` you can also use
`HTML:` to directly print the HTML-nodes...

* width: <script output="width" default="30" input="range" run-once>@input</script> %
* bottom: <script output="bottom" default="30" input="range" run-once>@input</script> %
* left: <script output="left" default="30" input="range" run-once>@input</script> %

<script style="display: block" modify="false">
`LIASCRIPT: 
![Kitten](https://quarto.org/docs/presentations/revealjs/demo/mini/images/kitten-400-350.jpeg)<!--
style="width: @input(`width`)%; position: absolute; bottom: @input(`bottom`)%; left: @input(`left`)%"
-->`
</script>
```

**Reply by abotzki** [reply]
Thanks a lot  @andre-dietrich ! I am happy to see that there are so many options! 

For an ELIXIR training course about FAIR training material design, I am preparing a [small example presentation](https://liascript.github.io/course/?https://raw.githubusercontent.com/vibbits/material-liascript/master/example-presentation.md#23) (still WIP ;-)) based on the [demo slides](https://quarto.org/docs/presentations/revealjs/demo/#/title-slide) which the quarto team has developed for their framework. 

The idea is to provide examples from several tools to create OERs including slides. And most importantly also demostrate that Liascript is a very versatile tool you could do a lot of more stuff with.

**Reply by andre-dietrich** [reply]
Ow, that sounds great ;-) don't forget to take a look at the templates:

https://github.com/topics/liascript-template

I always prefer ABCjs as an alternative example to live programming

https://github.com/LiaTemplates/ABCjs

**Reply by andre-dietrich** [reply]
By the way, as a new option for quizzes, you can also combine ASCII-art with LiaScript ... Quotes are used to define a verbatim environment, which means, within the ASCII-art image, you can also embed any kind of LiaScript (including animations)... To trick the interpreter into Gap-Text instead of a single-line quiz, the input field has to be surrounded by spaces ... 

```` markdown
``` ascii
                               Sum-Pyramid
                        .----------------------.
                       /                      /|
             .--------+----------------------+ +---------.
            /         |      " [[  24   ]] " |/         /|
  .--------+----------+----------+-----------+---------+ +----------.
 /         |         11          |     " [[   13   ]] "|/          /|
+----------+----------+----------+----------+----------+----------+ +
|      " [[   5   ]] "|          6          |          7          |/
+---------------------+---------------------+---------------------+
```
````
See a working example [here](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA51UbW/TMBD+7l9xayW0rXJMtwHaNE1ME4J+ZnyqqtZ1TGOIX2Rfuk3qj8dOwtquCQNOkXy+8/l5fC8Zwmfu6L18RHgDt1/vJhN665GQiQnSI2AhQVTeS4FgKr2UPkCoRBEdvPG6J8+1ysF5u1a5TG4d4Lu3GpYWMS5o4+eyLCPk+ohSyDlyGgr7QB33qHhJgy0rVNYApTdksVgAD0IpAj2S0U7J+gJYj3mzH/B87aj7/hFsHftYW4BNswxgOgU4u4jqbBZ3m+2JGrULqlvd0XfxE4ED1CjjMby0DhIVGJ+3XHaoJC6vEnhdhRHZeza8+w21RXp/QCvJhx0rI91p/ydrah6SZDiEieYrWXccUXp1BUfT2XGB6MIVY5UrLc+zB/VTaZkrnlm/Ymnn0o4Jq7U1gWERW55dskvBvlSam/mnJzm/99yEdRwEOb+rcP7NlHwpS5lnP9yKjd+eXbhH+nenT4Ckhv/PqRjAgXyMDz20tsdjVur6HJfShJNYButQCchVEHW1ni8+7cxtIto7kT1y3UTWHRhEKT1vGqMuUcvGS1SGJz6isD7+SHq53NRRf8Q6bTuweZqRfi2hhYwJ/AUErGf67AQAAA==)

Just an idea ;-)

**Reply by abotzki** [reply]
Very nice, the example with the eye is excellent!
